tsnet/example/tsnet-funnel/tsnet-funnel.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// The tsnet-funnel server demonstrates how to use tsnet with funnel.
```

Comment:
There is some ideal world where you write a tsnet test that starts a testcontrol and then builds+executes this example against it. I added testcontrol-based tests of tsnet. Might be worth a try.

Reply:
I couldn't figure out a nice way to do this for the funnel bit, I'll add it as a follow up.

Reply:
https://github.com/tailscale/tailscale/pull/7519